### ISSUE TITLE
Windows Outlook: internetHeaders.setAsync being ignored in on-send hook

### ISSUE THREAD

**Issue by carlosb1504** [OfficeDev/office-js#2952]
**PLEASE NOTE**: The behaviour described here is exactly the same as what is described in a ticket from December 2020, here: https://github.com/OfficeDev/office-js/issues/1401 - possible regression?

When you have a function file that processes a message during the send event any attempt to set internet headers during this window, specifically on Outlook for Windows, seems to result in a success from Office.js, in so much that no error is return in the async result, but when the recipient inspects the message headers, the specified header is not there.

Setting headers from a taskpane works as expected and these are present on the message in the recipients mailbox.

**Expected Behavior**
Headers that are set during the send event processing will persist and be sent with the message once the event has completed.

**Current Behavior**
Calling the setAsync methods returns as successful but the header is not persisted when the message arrives in the recipients mailbox.

Calling saveAsync on the item after setting the header seems to make no difference as well.

When the same code is run in either OWA or Outlook for Mac, the headers are present as expected when the message arrives in the recipients mailbox.

**Steps to Reproduce, or Live Example**
https://github.com/jak-hammond/office-js-on-send/tree/message-headers

**Additional details:**
Clone repo and install manifest
Create a new message
Open the taskpane and click Run
Send the message
Inspect the message properties; File -> Properties -> Internet Headers
Find 'x-taskpane-header' and x-onsend-header'
**Context**
We run some checks when a message and based on our results, we set various message headers that can be used later on in a users mail flow, at present, this means we cannot support Windows Outlook users unless they are aware they need to open the taskpane first which is a subpar user experience and can interrupt their workflow, the on-send hook should be able to handle this all silently for each send.

**Your Environment**
Platform [PC desktop, Mac, iOS, Office on the web]: PC Desktop
Host [Excel, Word, PowerPoint, etc.]: Outlook
Office version number: Version 2210 (Build 15726.20174 Click-to-Run) Current Channel
Operating System: Windows 10
Browser (if using Office on the web): Successful functionality can be seen in OWA using Chrome

**Comment by carlosb1504** [comment]
Good morning, getting on for 4 days since I logged this one with no reply. It's causing our users issues in production as we rely on these internet headers being set for certain elements of our add-in's functionality.

Any updates please?

**Comment by exextoc** [comment]
This has found to be a regression from a change from a few months ago. We are looking to see if it is possible to roll out an emergency fix instead of waiting for the next build. We will update when we know more.

**Comment by carlosb1504** [comment]
Thanks for the update. Is there a way that we can include the last unaffected version of the scripts in our application until this issue is resolved?

**Comment by exextoc** [comment]
The last unaffected version should be 16.0.15629.XXXXX 

An emergency fix is being rolled out, and is scheduled to hit most builds in a few days. This schedule is not final, and can change if issues are found. I will update as more information is available.

**Comment by exextoc** [comment]
The fix has been rolled out to Monthly Channels (builds 15726 and 15831) The Outlook Clients will have to be rebooted and connected to the internet. No, download/update is required. The fix should be live now, but it may be good to wait a few days to be safe.

**Comment by carlosb1504** [comment]
Thanks for the update, @exextoc, looks like we're starting to see the fix rolling out now.

Given that an Outlook update wasn't required in this case, presumably the fix was in the Office.JS Javascript itself.

A question I've asked before but not had answered. In these situations where a breaking change has been made is it possible for us as developers to include a 'last known working' version of the script to as to mitigate the problems for our users more quickly? Then once the MS fix has been deployed, move back to the latest version? Seems to be that this would be far preferable as we aren't having to wait for a fix to be identified, tested, released and then propagated worldwide. Any thoughts on this? Thanks

Also Cc: @JuaneloJuanelo, @ajays-msft

**Comment by exextoc** [comment]
Actually, this wasn't an Office JS issue. It was a bug in the client. We had a rework of the Item Send code that rolled out recently, that shouldn't have impacted end users, but was necessary code rework for maintenance and groundwork for future fixes. Unfortunately, we did not catch this issue during testing. Office clients have the ability to "roll back/turn off" client side fixes without requiring a full download in certain situations and channels. In this case, this bug was caught earlier enough that we were able to catch it and deploy this type of fix.

I believe that the office.js is available to download and self host, but doing so also opens you up to not getting fixes/etc.
